src/App.spec.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { render, fireEvent, waitFor } from '@testing-library/react';
+import App from './App';
+
+const WAIT_TIMEOUT_MS = 3000;
+
+const apiConfig = { BATCH_DUMP_TIMEOUT_MS: 1, BATCH_URL: '/file-batch-api' };
+jest.mock('./api-client/consts', () => apiConfig);
+
+describe('App component', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('fetches and displays files correctly', async () => {
```

Comment:
This could use a test where we could see that everything works with multiple batch dumps.
But I was lazy. Spent quite a lot of time.

Reply:
If it were production code I'd have added such a test.